pkg/datapath/loader/base.go

```diff
@@ -363,6 +364,15 @@ func (l *Loader) Reinitialize(ctx context.Context, o datapath.BaseProgramOwner,
 		args[initArgTunnelMode] = option.TunnelVXLAN
 	}
 
+	switch args[initArgTunnelMode] {
+	case option.TunnelVXLAN:
+		args[initArgTunnelPort] = fmt.Sprintf("%d", option.TunnelPortDefaultVXLAN)
+	case option.TunnelGeneve:
+		args[initArgTunnelPort] = fmt.Sprintf("%d", option.TunnelPortDefaultGeneve)
+	default:
+		args[initArgTunnelPort] = "<nil>"
```

Comment:
Hm, is it me or is the value from `--tunnel-port` never used?

We can also get rid of the `<nil>` case by defining `args[initArgTunnelPort]` to have the default `option.TunnelPortDefaultXXX` value if `--tunnel-port` isn't set.

Reply:
Haha, thanks! Well spotted ;)

Reply:
> We can also get rid of the `<nil>` case by defining `args[initArgTunnelPort]` to have the default `option.TunnelPortDefaultXXX` value if `--tunnel-port` isn't set.

The point was set to `<nil>` in case if somehow a new tunnel type is added, right now I'm not seeing checks for unknown values anywhere, so it seems theoretically possible...

Reply:
Spoke too soon, there is this error condition: https://github.com/cilium/cilium/blob/fb85fee7001e7fcc9fc3466c4ff018fe787fe26c/pkg/option/config.go#L2202-L2204